jctools-core/src/main/java/org/jctools/queues/AbstractSpscResizingArrayQueue.java

```diff
@@ -0,0 +1,105 @@
+package org.jctools.queues;
+
+import java.lang.reflect.Field;
+import java.util.AbstractQueue;
+import java.util.Iterator;
+
+import static org.jctools.util.UnsafeAccess.UNSAFE;
+
+abstract class SpscResizingArrayQueuePrePad<E> extends AbstractQueue<E> {
+    long p0, p1, p2, p3, p4, p5, p6, p7;
+    long p10, p11, p12;
+    // p13, p14, p15, p16, p17; drop 4 longs, the cold fields act as buffer
+}
+abstract class SpscResizingArrayQueueProducerColdFields<E> extends SpscResizingArrayQueuePrePad<E> {
+    protected int maxQueueCapacity; // ignored by the unbounded implementation
+    protected int producerLookAheadStep;
+    protected long producerLookAhead;
+    protected long producerMask;
+    protected E[] producerBuffer;
+}
+abstract class SpscResizingArrayQueueProducerFields<E> extends SpscResizingArrayQueueProducerColdFields<E> {
+    protected long producerIndex;
+}
+abstract class SpscResizingArrayQueueL2Pad<E> extends SpscResizingArrayQueueProducerFields<E> {
+    long p0, p1, p2, p3, p4, p5, p6, p7;
+    long p10, p11, p12, p13, p14, p15, p16, p17;
+}
+abstract class SpscResizingArrayQueueConsumerColdFields<E> extends SpscResizingArrayQueueL2Pad<E> {
+    protected long consumerMask;
+    protected E[] consumerBuffer;
+}
+abstract class SpscResizingArrayQueueConsumerField<E> extends SpscResizingArrayQueueConsumerColdFields<E> {
+    protected long consumerIndex;
+}
+
+abstract class AbstractSpscResizingArrayQueue<E> extends SpscResizingArrayQueueConsumerField<E>
```

Comment:
Perhaps Abstract or Base SpscLinkedArrayQueue is a better name? not sure, just my gut feeling